src/Features/CSharp/Portable/BraceCompletion/AbstractCurlyBraceOrBracketCompletionService.cs

```diff
@@ -115,18 +114,24 @@ private static bool ContainsOnlyWhitespace(SourceText text, int openingPosition,
             if (closingPointLine - openingPointLine == 1)
             {
                 // Handling syntax tree directly to avoid parsing in potentially UI blocking code-path
-                var closingToken = document.Root.FindTokenOnLeftOfPosition(context.ClosingPoint);
-                var newLineString = options.FormattingOptions.NewLine;
-                newLineEdit = new TextChange(new TextSpan(closingToken.FullSpan.Start, 0), newLineString);
+                var closingToken = document.Root.FindToken(closingPoint - 1);
+                Debug.Assert(IsValidClosingBraceToken(closingToken));
 
-                var generator = document.LanguageServices.GetRequiredService<SyntaxGeneratorInternal>();
-                var endOfLine = generator.EndOfLine(newLineString);
+                var newClosingToken = closingToken.WithPrependedLeadingTrivia(
+                    SpecializedCollections.SingletonEnumerable(SyntaxFactory.EndOfLine(options.FormattingOptions.NewLine)));
```

Comment:
note: this generates a potentially invalid tree (i.e. something different than what the parser would produce).  That may be fine, but it's something worth thining about in base it breaks other layers.

Reply:
We only need this change to get through the call to formatter within the method, since we return text change only for brace completion session, not parsed tree. Which I assume will be directly applied to the buffer, any update to current solution will happen in the background after that.

Reply:
wfm.  though it's still iffy as i could imagine the formatter syaing that it doesn't expect this.  hopefully this won't bite us later :)